ENGAGEHF/Home.swift

```diff
@@ -18,27 +20,43 @@ struct HomeView: View {
     static var accountEnabled: Bool {
         !FeatureFlags.disableFirebase && !FeatureFlags.skipOnboarding
     }
+    
+    private var bluetoothEnabled: Bool {
+        !ProcessInfo.processInfo.isPreviewSimulator
+    }
 
-
+    @Environment(MeasurementManager.self) private var measurementManager
+    @Environment(WeightScaleDevice.self) private var weightScale: WeightScaleDevice?
+    @Environment(Bluetooth.self) private var bluetooth
+    
     @AppStorage(StorageKeys.homeTabSelection) private var selectedTab = Tabs.home
     @State private var presentingAccount = false
-
+    
     
     var body: some View {
+        @Bindable var measurementManager = measurementManager
+        
         TabView(selection: $selectedTab) {
             Dashboard(presentingAccount: $presentingAccount)
                 .tag(Tabs.home)
                 .tabItem {
                     Label("Home", systemImage: "house")
                 }
         }
+            .autoConnect(enabled: weightScale == nil && bluetoothEnabled, with: bluetooth)
```

Comment:
I am not quire sure that the intention behind the `weightScale` input here is.
Just to be sure, it might be to disable autoconnect if a scale is already connected?

if so, it might make sense to move this in the computed property and rename it to `autoconnect`? It would also make sense to add a documentation about this there 👍

Reply:
If I add some context on the behavior of the auto connect modifier. If enabled, it searches till it finds the first device it can connect to. Checking for `weightScale == nil` is not necessary. Autoconnect won't continue scanning if there is already a device connected.
In NAMS we had a similar check, however there it was required as we had devices that were not managed by SpeziBluetooth (e.g., so we didn't want to auto connect another device if there was a device connected via a different mechanisms).
Not a big deal, but might make the code a little simpler 👍

Reply:
Thank you for the context @Supereg , that makes things easier 🚀